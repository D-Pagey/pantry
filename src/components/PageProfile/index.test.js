import React from 'react';
import userEvent from '@testing-library/user-event';
import { PageProfile } from '.';

const props = {};

const context = {
  categories: [{ label: 'meat', colour: 'red' }],
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  signOut: () => {},
  user: {
    email: 'dan@gmail.com',
    name: 'Dan',
  },
};

describe('PageProfile component', () => {
  it('should render', () => {
    const { container } = render(<PageProfile {...props} />, context);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call sign out on click', () => {
    const signOut = jest.fn();
    const { getByTestId } = render(<PageProfile {...props} />, { ...context, signOut });

    userEvent.click(getByTestId('pageProfileButton'));

    expect(signOut).toHaveBeenCalled();
  });
});
