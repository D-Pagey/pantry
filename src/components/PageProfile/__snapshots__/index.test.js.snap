// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageProfile component should render 1`] = `
.c0 {
  background-color: red;
  height: 20px;
  width: 20px;
}

<div
  data-testid="pageProfile"
>
  <p>
    Welcome 
    Dan
  </p>
  <p>
    Your email is: 
    dan@gmail.com
  </p>
  <button
    data-testid="pageProfileButton"
  >
    Sign Out
  </button>
  <h2>
    Your Food Categories
  </h2>
  <div
    class="ReactTable"
  >
    <div
      class="rt-table"
      role="grid"
    >
      <div
        class="rt-thead -header"
        style="min-width: 200px;"
      >
        <div
          class="rt-tr"
          role="row"
        >
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Category
            </div>
            <div
              class="rt-resizer"
            />
          </div>
          <div
            class="rt-th rt-resizable-header -cursor-pointer"
            role="columnheader"
            style="flex: 100 0 auto; width: 100px;"
            tabindex="-1"
          >
            <div
              class="rt-resizable-header-content"
            >
              Colour
            </div>
            <div
              class="rt-resizer"
            />
          </div>
        </div>
      </div>
      <div
        class="rt-tbody"
        style="min-width: 200px;"
      >
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              meat
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <div
                class="c0"
              />
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -odd"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
        <div
          class="rt-tr-group"
          role="rowgroup"
        >
          <div
            class="rt-tr -padRow -even"
            role="row"
          >
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
            <div
              class="rt-td"
              role="gridcell"
              style="flex: 100 0 auto; width: 100px;"
            >
              <span>
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination-bottom"
    >
      <div
        class="-pagination"
      >
        <div
          class="-previous"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Previous
          </button>
        </div>
        <div
          class="-center"
        >
          <span
            class="-pageInfo"
          >
            Page
             
            <div
              class="-pageJump"
            >
              <input
                aria-label="jump to page"
                type="number"
                value="1"
              />
            </div>
             
            of
             
            <span
              class="-totalPages"
            >
              1
            </span>
          </span>
          <span
            class="select-wrap -pageSizeOptions"
          >
            <select
              aria-label="rows per page"
            >
              <option
                value="5"
              >
                5 rows
              </option>
              <option
                value="10"
              >
                10 rows
              </option>
              <option
                value="20"
              >
                20 rows
              </option>
              <option
                value="25"
              >
                25 rows
              </option>
              <option
                value="50"
              >
                50 rows
              </option>
              <option
                value="100"
              >
                100 rows
              </option>
            </select>
          </span>
        </div>
        <div
          class="-next"
        >
          <button
            class="-btn"
            disabled=""
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="-loading"
    >
      <div
        class="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
